// components/mailhelper.js
//Import the nodemailer package (npm install nodemailer)
const nodemailer = require("nodemailer");

//Configure the mail Broker with OAuth2
  mailTransport = nodemailer.createTransport({
    host: ""/*Mail Broker e.g. smtp.gmail.com */,
    service: ""/*Mail Client*/,
    secure: false,
    auth: {
      type: 'OAuth2',
      user: ""/*Username */,
      clientId: ""/*OAUTH Client ID */,
      clientSecret: "" /*Client Secret*/,
      refreshToken: ""/*Refresh Token*/,
      accessToken: ""/*Access Token */
    },
    tls: {
      rejectUnauthorized: false
  }
  });

  //Function to send Email
  exports.sendMail = function (maildetails) {
      //Error Handling using promise
    return new Promise(
        function(resolve,reject) {
            //Send Mail using mailTransport
    mailTransport.sendMail(maildetails, (err, data) => {
      if (err) {
        console.log("Error" + err);
        //Promise recalled
        reject(err);
      } else {
          //Successfully sent email
        console.log("Successfully sent to " + maildetails.to);
        resolve("success");
      }
    })
    });
  
  };
